### PR TITLE
Isolated benchmarking support

### PR DESCRIPTION
**Pull request by anirudh2290**
Allows for benchmarking only IO or compute costs using --io-only and --compute-only flags and benchmarking communication costs using --communication-only
@eric-haibin-lin @reminisce @piiswrong

### CODE REVIEW

File: benchmark/python/sparse/sparse_end2end.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def row_sparse_pull(kv, key, data, slices, weight_array, priority):
     compute_only = args.compute_only
     communication_only = args.communication_only
     if (compute_only and io_only) or (compute_only and communication_only) or (io_only and communication_only):
-        raise Exception("Only one of compute_only, io_only, communication_only can be set")
+        assert False, "Only one of compute_only, io_only, communication_only can be set"
```
Comment (szha) [inline benchmark/python/sparse/sparse_end2end.py]: `assert compute_only + io_only + communication_only <= 1`

Comment (anirudh2290) [reply benchmark/python/sparse/sparse_end2end.py]: Thank you for the suggestion! Changed.

File: benchmark/python/sparse/sparse_end2end.py [hunk 2]
```diff
@@ -140,11 +166,24 @@ def row_sparse_pull(kv, key, data, slices, weight_array, priority):
...
     dataset = args.dataset
     log_level = args.sparse_log_level
+    io_only = args.io_only
+    compute_only = args.compute_only
+    communication_only = args.communication_only
+    num_cores = multiprocessing.cpu_count()
+    assert (compute_only + io_only + communication_only <= 1), "Only one of compute_only, io_only, communication_only can be set"
+    if compute_only or io_only:
+        assert not kvstore, "when compute_only or io_only is set, kvstore should be None"
+        num_batch = datasets[dataset]['lc'] / batch_size if num_batch == 99999999 else num_batch
```
Comment (eric-haibin-lin) [inline benchmark/python/sparse/sparse_end2end.py]: replace 99999999  with macro 
num_batch =  MAX_NUM_BATCH

File: benchmark/python/sparse/sparse_end2end.py [hunk 3]
```diff
@@ -48,6 +49,12 @@
                     help='whether to call update_metric')
 parser.add_argument('--enable-logging-for', default="0",
                     help="Enable logging for the specified list of workers")
+parser.add_argument('--io-only', action='store_true',
```
Comment (eric-haibin-lin) [inline benchmark/python/sparse/sparse_end2end.py]: I actually prefer one option for `measure_only = IO/COMP/COMM, default to None`

File: benchmark/python/sparse/sparse_end2end.py [hunk 4]
```diff
@@ -181,7 +220,7 @@ def row_sparse_pull(kv, key, data, slices, weight_array, priority):
     train_data = mx.io.LibSVMIter(data_libsvm=path, data_shape=(feature_dim,),
                                   batch_size=batch_size, num_parts=num_worker,
                                   part_index=rank)
-    if dummy_iter:
+    if dummy_iter or compute_only:
```
Comment (eric-haibin-lin) [inline benchmark/python/sparse/sparse_end2end.py]: Is DummyIter not used for communication_only????

File: benchmark/python/sparse/sparse_end2end.py (hunk 3 above)
Comment (anirudh2290) [reply benchmark/python/sparse/sparse_end2end.py]: Changed.

File: benchmark/python/sparse/sparse_end2end.py (hunk 2 above)
Comment (anirudh2290) [reply benchmark/python/sparse/sparse_end2end.py]: Changed.

File: benchmark/python/sparse/sparse_end2end.py (hunk 4 above)
Comment (anirudh2290) [reply benchmark/python/sparse/sparse_end2end.py]: Nice catch. This was a mistake. Added communication_only.

**Pull request merged**